ilc/client/navigationEvents/setupEvents.js

```diff
@@ -18,7 +26,24 @@ function patchedUpdateState(updateState) {
     };
 }
 
-// We will trigger an app change for any routing events.
+let currentUrl = window.location.href;
+
+function handlePopState(event) {
+    const oldUrl = currentUrl;
+    const newUrl = (currentUrl = window.location.href);
+
+    if (event.singleSpa || shouldNavigateTo(newUrl)) {
+        window.callCapturedEventListeners(arguments);
+        return;
+    }
+
+    window.history.replaceState(null, undefined, oldUrl);
```

Comment:
Now, I have decided to use `history.replaceState` here to change the current `location.href` because we do not have an opportunity to check what exactly button would be clicked by a user: forward, back, or an entry in History.